pkg/sockets/tcp_socket.go

```diff
@@ -4,11 +4,56 @@ package sockets
 
 import (
 	"crypto/tls"
+	"fmt"
 	"net"
 	"net/http"
+	"net/url"
+	"os"
+	"strings"
 	"time"
+
+	"golang.org/x/net/proxy"
 )
 
+// The uppercase and the lowercase are available for the proxy settings.
+// See the Go specification for details on these variables. https://golang.org/pkg/net/http/
+func GetProxyEnv(key string) string {
+	proxyValue := os.Getenv(strings.ToUpper(key))
+	if proxyValue == "" {
+		return os.Getenv(strings.ToLower(key))
+	}
+	return proxyValue
+}
+
+func DialerFromEnvironment(timeout time.Duration) (proxy.Dialer, error) {
```

Comment:
This is duplicated-ish from the Golang package itself so that we can get the custom timeout and actually return errors when they come up.  Hopefully can contribute upstream and get this back to using `DialerFromEnvironment` from the `proxy` package directly.